crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -1832,6 +1833,54 @@ impl Store {
         from_backup_version: Option<&str>,
         progress_listener: impl Fn(usize, usize),
     ) -> Result<RoomKeyImportResult> {
+        let exported_keys: Vec<&ExportedRoomKey> = exported_keys.iter().collect();
+        self.import_sessions_impl(exported_keys, from_backup_version, progress_listener).await
+    }
+
+    /// Import the given room keys into our store.
+    ///
+    /// # Arguments
+    ///
+    /// * `exported_keys` - A list of previously exported keys that should be
+    ///   imported into our store. If we already have a better version of a key
+    ///   the key will *not* be imported.
+    ///
+    /// Returns a tuple of numbers that represent the number of sessions that
+    /// were imported and the total number of sessions that were found in the
+    /// key export.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use std::io::Cursor;
+    /// # use matrix_sdk_crypto::{OlmMachine, decrypt_room_key_export};
+    /// # use ruma::{device_id, user_id};
+    /// # let alice = user_id!("@alice:example.org");
+    /// # async {
+    /// # let machine = OlmMachine::new(&alice, device_id!("DEVICEID")).await;
+    /// # let export = Cursor::new("".to_owned());
+    /// let exported_keys = decrypt_room_key_export(export, "1234").unwrap();
+    /// machine.store().import_exported_room_keys(exported_keys, |_, _| {}).await.unwrap();
+    /// # };
+    /// ```
+    pub async fn import_exported_room_keys(
+        &self,
+        exported_keys: Vec<ExportedRoomKey>,
+        progress_listener: impl Fn(usize, usize),
+    ) -> Result<RoomKeyImportResult> {
+        self.import_room_keys(exported_keys, None, progress_listener).await
+    }
+
+    async fn import_sessions_impl<T>(
+        &self,
+        room_keys: Vec<T>,
```

Comment:
Can this be `IntoIter<T>` or something, to avoid the `collect()` above when we call it?

Reply:
Kinda, though it can't be just that, because we have a `room_keys.len()` call.

Which then makes all of the trait bounds even more complex.

The other option is to remove the `len()` call, but then our progress listener doesn't work anymore.

Don't think it's worth it to avoid this `collect()` call, after all, allocating a vector of references isn't that big of a deal.

Reply:
Yep, agree it's fine as-is.

Reply:
(Or pass in the length?)

Reply:
> (Or pass in the length?)

That sounds like it would be a bit of a clunky API, not terrible but clunky enough for me to rather take the allocation.

The only reason why we take a `Vec<&Foo>` vs a `Vec<Foo>` is so we can gather info about the specific key in the error branch if creating a `InboundGroupSession` fails. 

Perhaps gathering the info before the `try_into()` call consumes the key would be the wiser choice if we really want to avoid the allocation.

Reply:
Alright, let's keep the allocation, if we change our mind I think gathering the info before the consumption of the key would be the way to go.